src/app/utils/pinecone.ts

```diff
@@ -21,25 +20,34 @@ const getMatchesFromEmbeddings = async (embeddings: number[], topK: number, name
   // Retrieve the list of indexes
   const indexes = await pinecone.listIndexes()
 
+  let exists = false
+  for (const index of indexes) {
+    if (index.name === process.env.PINECONE_INDEX!) {
+      exists = true
+    }
+  }
+
   // Check if the desired index is present, else throw an error
-  if (!indexes.includes(process.env.PINECONE_INDEX!)) {
+  if (!exists) {
     throw (new Error(`Index ${process.env.PINECONE_INDEX} does not exist`))
   }
 
   // Get the Pinecone index
   const index = pinecone!.Index(process.env.PINECONE_INDEX!);
 
+  // Get the namespace
+  const pineconeNamespace = index.namespace(namespace ?? '')
+
   // Define the query request
-  const queryRequest = {
+  const queryOptions = {
```

Comment:
This looks correct, but from a style perspective I would probably just define this object inline with the call to query. I think the old way of defining `queryRequest` and then passing `{ queryRequest }` was mainly to obfuscate how silly it was having that top-level `queryRequest` object key.

```typescript
const queryResult = await pineconeNamespace.query({
  vector: embeddings,
  topK,
  includeMetadata
})
```

Reply:
Yeah, that is much cleaner. Done in 14650f5.